cmd/builder/internal/version.go

```diff
@@ -14,18 +14,34 @@ var (
 	version = ""
 )
 
-func init() {
-	// the second returned value is a boolean, which is true if the binaries are built with module support.
-	info, _ := debug.ReadBuildInfo()
-	version = info.Main.Version
+type debugReadBuildInfoFunc func() (info *debug.BuildInfo, ok bool)
+type binVersionFunc func(fn debugReadBuildInfoFunc) (string, error)
+
+// binVersion returns the version of the binary.
+// If the version is not set, it attempts to read the build information.
+// Returns an error if the build information cannot be read.
+func binVersion(fn debugReadBuildInfoFunc) (string, error) {
+	if version != "" {
+		return version, nil
+	}
+	info, ok := fn()
+	if !ok {
+		return "", fmt.Errorf("failed to read build info")
+	}
+	return info.Main.Version, nil
 }
 
-func versionCommand() *cobra.Command {
+func versionCommand(fn binVersionFunc) *cobra.Command {
+	var err error
 	return &cobra.Command{
 		Use:   "version",
 		Short: "Version of ocb",
 		Long:  "Prints the version of the ocb binary",
 		RunE: func(cmd *cobra.Command, _ []string) error {
+			version, err = fn(debug.ReadBuildInfo)
+			if err != nil {
+				return err
+			}
```

Comment:
Is the difference just that the `debug.ReadBuildInfo()` is executed in the Run? If that is the case, can we do just that?

Reply:
no, that's not enough to make a difference. running `CGO_ENABLED=0 go build -trimpath -ldflags='-s -w -X go.opentelemetry.io/collector/cmd/builder/internal.version=v0.110.0'` with this code passes a version number but just moving the debug.ReadBuildInfo into the RunE command does not fix it.

Reply:
need to have a check to see if version is still blank; the init function is overwriting version with `debug.ReadBuildInfo` otherwise

Reply:
simply adding an `if version == ""` check to the `init()` function originally on line 17 would also solve it, it seems

Reply:
ok @bogdandrutu I made the PR much smaller and still fixed the issue

Reply:
>running CGO_ENABLED=0 go build -trimpath -ldflags='-s -w -X go.opentelemetry.io/collector/cmd/builder/internal.version=v0.110.0' 

Where do we do this?